public_html/wp-content/mu-plugins/blocks/assets/src/blocks-store.js

```diff
@@ -0,0 +1,199 @@
+/**
+ * External dependencies
+ */
+import createSelector from 'rememo';
+
+/**
+ * WordPress dependencies
+ */
+const { registerStore, dispatch } = wp.data;
+const { apiFetch }                = wp;
+const { addQueryArgs }            = wp.url;
+
+/**
+ * Define store name.
+ */
+export const WC_BLOCKS_STORE = 'wc-blocks-store';
+
+/**
+ * Initial state.
+ */
+const DEFAULT_STATE = {
+	loadingEntities: [],
+};
+
+const MAX_POSTS = 100;
+
+/**
+ * Generic query for supported data type.
+ */
+const API_ARGS = {
+	entity: {
+		path  : ( entityType ) => 'wp/v2/' + entityType,
+		query : {
+			orderby  : 'id',
+			order    : 'desc',
+			per_page : MAX_POSTS,
+			_embed   : true,
+			context  : 'view',
+		},
+	},
+};
+
+/**
+ * Helper method for fetching WordCamp Custom Entities.
+ *
+ * @param {Object} state      Current store state
+ * @param {string} entityType Name of the entity. Egs post, speaker etc
+ * @param {string} path       REST API path to fetch entity records
+ * @param {Object} query      Query params for the REST API request
+ */
+const apiFetchEntities = ( state, entityType, path, query ) => {
+	// Bail if we already have these entities fetched.
+	if ( state.hasOwnProperty( entityType ) && 0 !== state[ entityType ].length ) {
+		return;
+	}
+
+	// Bail if already loading this entity
+	if ( -1 !== state.loadingEntities.indexOf( entityType ) ) {
+		return;
+	}
+
+	state.loadingEntities.push( entityType );
+
+	// TODO: Implement pagination.
+	const apiFetchResult = apiFetch( {
+		path: addQueryArgs( path, query ),
+	} );
+
+	apiFetchResult.then(
+		( customEntities ) => {
+			dispatch( WC_BLOCKS_STORE ).setEntities( entityType, customEntities );
+		}
+	).catch( //TODO: Implement retries on HTTP Transport errors.
+		( reason ) => {
+			console.error( entityType, 'Unable to retrieve data from API.', reason );
+		}
+	);
+};
+
+/**
+ * Define actions which can be dispatched by this store.
+ *
+ * @type {Object}
+ */
+const actions = {
+	/**
+	 * Queues fetching from API for a post type.
+	 *
+	 * @param {string} entityType
+	 * @returns {Object}
+	 */
+	fetchEntities( entityType ) {
+		return {
+			type       : 'FETCH_ENTITIES',
+			entityType : entityType,
+		};
+	},
+
+	/**
+	 * Set entities state
+	 *
+	 * @param {string} entityType
+	 * @param {Array}  entities
+	 *
+	 * @return {Object}
+	 */
+	setEntities( entityType, entities ) {
+		return {
+			type       : 'SET_ENTITIES',
+			entityType : entityType,
+			entities   : entities,
+		};
+	},
+};
+
+/**
+ * Defines selectors provided by this store.
+ *
+ * @type {{getPosts(*, *)}}
+ */
+const selectors = {
+
+	/**
+	 * Returns post type from current state. Caches [state, entityType] for quick resolution.
+	 *
+	 * @param {Object} state      Current store state
+	 * @param {string} entityType Name of the entity to get
+	 * @param {Object} args       Additional filter arguments.
+	 */
+	getEntities: createSelector(
+		( state, entityType, args = {} ) => {
```

Comment:
We have removed this now that we have moved to core store's `getEntityRecords`. 

That said, this is how Gutenberg core also uses `createSelector`, where first param is the actual selector function, and second param just returns the dependents. Selector will be called only when dependents value have been changed, otherwise cached value from last iteration will be returned.

Reply:
Would moving the first param into a named function change how the caching occurs? My comment wasn't intended to change any functionality, just make the code easier to read.

Reply:
Ah ok. Yeah we could have used a named function here.

Reply:
Do you think it would help or hurt readability if we used a named function? If it'd help, is there any reason not to?

Reply:
Er, nevermind, I forgot the earlier comment where you said it was removed :)

Reply:
It looks like this is resolved? Closing, but please reopen if I am mistaken.

Reply:
👍